relay-server/src/actors/processor.rs

```diff
@@ -1926,8 +1930,70 @@ impl EnvelopeProcessorService {
             path: "".to_owned(),
             project_id: Some(scoping.project_id),
         };
-        for (partition_key, buckets) in partition_buckets(scoping.project_key, buckets, partitions)
+
+        let bucket_qty = buckets.len();
+        let bucket_partitions = partition_buckets(scoping.project_key, buckets, partitions);
+
+        let quotas = project_state.config.quotas.as_slice();
+        let item_scoping = ItemScoping {
+            category: DataCategory::Metrics,
+            scoping: &scoping,
+        };
+
+        if cached_rate_limits
```

Comment:
I think we should move all the rate limit checks to a separate function, this `handle_encode_metrics` is getting quite crowded.